src/Microsoft.DotNet.XHarness.CLI/Commands/WASM/JS/WasmTestCommand.cs

```diff
@@ -177,6 +179,11 @@ protected override async Task<ExitCode> InvokeInternal(ILogger logger)
             }
             else
             {
+                if(!wasmExitReceivedTcs.Task.IsCompletedSuccessfully)
+                {
+                    logger.LogError("Application exited without sending a wasm exit message.");
+                    return ExitCode.RETURN_CODE_NOT_SET;
+                }
```

Comment:
This should be before line 175, as it shouldn't depend on what `result.ExitCode` is.

Reply:
If v8 failed for any other reason, for example that test-main.js doesn't exist, I don't want it to be covered by message about WASM EXIT